core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -93,6 +93,16 @@ class SparkHadoopUtil extends Logging {
         hadoopConf.set("fs.s3.awsSecretAccessKey", accessKey)
         hadoopConf.set("fs.s3n.awsSecretAccessKey", accessKey)
         hadoopConf.set("fs.s3a.secret.key", accessKey)
+
+        if (System.getenv("AWS_SESSION_TOKEN") != null) {
+          val sessionToken = System.getenv("AWS_SESSION_TOKEN")
```

Comment:
You could get this value once, and then check it for null and also use its value. It would avoid duplicating the system prop name. You could also improve the similar check above.